cloud.foundry.cli/src/test/java/cloud/foundry/cli/logic/ApplyLogicTest.java

```diff
@@ -391,4 +385,110 @@ public void testApplyServices() {
+        applyLogic.applySpace(desiredSpaceName, spaceOperationsMock);
+
+        // then
+        verify(resultingMono).block();
+    }
+
+    @Test
+    public void testApplySpaceWithSpaceExistingWillNotCreateSpace() {
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        List<String> presentSpaces = Arrays.asList("testName", "space2");
+        when(spaceOperationsMock.getAll()).thenReturn(Mono.just(presentSpaces));
+
+        Mono<Void> resultingMono = mock(Mono.class);
+        when(spaceOperationsMock.create(desiredSpaceName)).thenReturn(resultingMono);
+
+        // when
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
+        applyLogic.applySpace(desiredSpaceName, spaceOperationsMock);
+
+        // then
+        verify(resultingMono, never()).block();
+    }
+
+    @Test
+    public void testApplySpaceWithGetSpaceNamesFailingThrowsGetException() {
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        Mono<List<String>> getRequestMock = mock(Mono.class);
+        when(spaceOperationsMock.getAll()).thenReturn(getRequestMock);
+        when(getRequestMock.block()).thenThrow(new RuntimeException("GetFailing"));
+
+        // when + then
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
+        assertThrows(GetException.class, () ->
+                applyLogic.applySpace(desiredSpaceName, spaceOperationsMock));
+    }
+
+    @Test
+    public void testApplySpaceWithCreateSpaceFailingThrowsApplyExcption() {
+
+        // given
+        String desiredSpaceName = "testName";
+        SpaceOperations spaceOperationsMock = mock(SpaceOperations.class);
+
+        List<String> presentSpaces = Arrays.asList("space1", "space2");
+        when(spaceOperationsMock.getAll()).thenReturn(Mono.just(presentSpaces));
+
+        Mono<Void> resultingMono = mock(Mono.class);
+        when(spaceOperationsMock.create(desiredSpaceName)).thenReturn(resultingMono);
+        when(resultingMono.block()).thenThrow(new RuntimeException("Create space failing"));
+
+        // when + then
+
+        // the constructor paramteres won't be used by apply space method, because it uses DI
+        // regarding space operations.
+        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
+        assertThrows(ApplyException.class, () ->
+                applyLogic.applySpace(desiredSpaceName, spaceOperationsMock));
```

Comment:
```suggestion
        // the constructor paramteres won't be used by apply space method, because it uses DI
        // regarding space operations.
        ApplyLogic applyLogic = new ApplyLogic(mock(DefaultCloudFoundryOperations.class));
         
         // when + then
        assertThrows(ApplyException.class, () ->
                applyLogic.applySpace(desiredSpaceName, spaceOperationsMock));
```